### PR TITLE
Allow setting SASL properties on the auto mesh connectors

### PR DESCRIPTION
**Pull request by ctron**
This changes allows to set the SASL properties on the connectors created by the auto mesh script.

### CODE REVIEW

**Comment by grs** [comment]
Can you give some detail on the use case for this? My instinct is to stick with TLS authentication as that provides the simplest way to restrict inter-router connections as distinct from normal connections (i.e. to essentially separate the user base of routers form normal users).

**Comment by ctron** [comment]
The use case is to create a router mesh, with auto discovery. Rather than rolling our own CA, I would like to re-use the OpenShift Service CA. However, the OpenShift Service CA does not create client certificates for X509 client auth. So we need another way of authenticating users.

**Comment by grs** [comment]
The cyrus sasl implementation doesn't allow separate user databases to be used and there is no good way at present to restrict the inter-router role to a specific subset of valid users. However if you only use cyrus sasl for authenticating connection the inter-router  listener, and use some other mechanism (e.g. the sasl delegation plugin) to authenticate connections to normal role listeners, that should work I guess.

**Comment by ctron** [comment]
I don't think we need a separate user database. This is some internal infrastructure functionality, which should not be exposed to any user. Basically we need two internal users. One for the inter-router mesh, and one for accessing the actual address. They are both managed by the operator, and can both reside in the same user database.

**Comment by grs** [comment]
The credentials for either will be valid for establishing inter-router connections. (i.e. though you can have two users, they are both entirely equivalent unless you have separate authorisation rules for them, and you can't authorise only one to be valid for inter-router connections).

**Comment by ctron** [comment]
So currently I put in an a vhost policy `$default`, which only allows two users, one with full access (the "interconnect" user) and one with access to only the single "IoT specific" address.

File: scripts/auto_mesh.py [hunk 1]
```diff
@@ -108,6 +108,17 @@ def get_connectors(hosts, port="55672", properties={}):
         c.update(properties)
     return connectors
 
+# add SASL properties, if they are set on the environment
+def add_sasl_properties(properties):
+    sasl_mechanisms = os.getenv('QDROUTERD_AUTO_MESH_SASL_MECHANISMS', 'PLAIN')
```
Comment (grs) [inline scripts/auto_mesh.py:113]: I would prefer to leave the sasl mechanisms unset if not explicitly provided I think. Other than that I think it is ok.

Comment (ctron) [reply scripts/auto_mesh.py]: Ok, I changed it.

**Pull request merged**